src/sdi-refresh-monitor.c

```diff
@@ -359,13 +360,12 @@ static void update_dock_bar(gpointer key, gpointer value, gpointer data) {
   g_variant_builder_add(builder, "{sv}", "updating",
                         g_variant_new_boolean(!task_data->done));
 
-  g_autoptr(GVariant) values =
-      g_object_ref_sink(g_variant_builder_end(builder));
+  g_autoptr(GVariant) values = g_variant_builder_end(builder);
 
   for (int i = 0; i < task_data->desktop_files->len; i++) {
     const gchar *desktop_file = task_data->desktop_files->pdata[i];
-    unity_com_canonical_unity_launcher_entry_emit_update(self->unity_manager,
-                                                         desktop_file, values);
+    unity_com_canonical_unity_launcher_entry_emit_update(
+        self->unity_manager, desktop_file, g_variant_ref(values));
```

Comment:
Why has this changed to passing a reference?

Reply:
Even worse: the program dies when updating two snaps due to triple-free, without the g_variant_ref.

Reply:
Looking at the generated code doesn't show it consuming the reference:
```
/**
 * unity_com_canonical_unity_launcher_entry_emit_update:
 * @object: A #UnityComCanonicalUnityLauncherEntry.
 * @arg_app_uri: Argument to pass with the signal.
 * @arg_properties: Argument to pass with the signal.
 *
 * Emits the <link linkend="gdbus-signal-com-canonical-Unity-LauncherEntry.Update">"Update"</link> D-Bus signal.
 */
void
unity_com_canonical_unity_launcher_entry_emit_update (
    UnityComCanonicalUnityLauncherEntry *object,
    const gchar *arg_app_uri,
    GVariant *arg_properties)
{ 
  g_signal_emit (object, UNITY__COM_CANONICAL_UNITY_LAUNCHER_ENTRY_SIGNALS[UNITY__COM_CANONICAL_UNITY_LAUNCHER_ENTRY_UPDATE], 0, arg_app_uri, arg_properties);
}
```

There must be something else going on here. If this is an existing problem it really should be in a separate PR, as this will be easier to diagnose later.

Reply:
I think the bug might be the `g_object_ref_sink`, which should have been `g_variant_ref_sink`.

Reply:
You are right. I didn't realize that GVariant had also sinkable refs. Fixed!